lib/iris/_shapefiles.py

```diff
@@ -0,0 +1,221 @@
+            "Shape is of invalid type for minimum weight masking, must use a Polygon rather than Line shape.\n Masking based off intersection instead. ",
+            IrisUserWarning,
+        )
+    else:
+        weights = True
+
+    # prepare shape
+    trans_geo = _transform_coord_system(geometry, cube, shape_coord_system)
+    prepped.prep(trans_geo)
+
+    # prepare 2D cube
+    y_name, x_name = _cube_primary_xy_coord_names(cube)
+    cube_2d = cube.slices([y_name, x_name]).next()
+    _cube_xy_guessbounds(cube_2d)
+    xmod = cube.coord(x_name).units.modulus
+    ymod = cube.coord(y_name).units.modulus
+    mask_template = np.zeros(cube_2d.shape, dtype=np.float64)
+
+    # perform the masking
+    for count, idx in enumerate(np.ndindex(cube_2d.shape)):
+        # get the bounds of the grid cell
+        xi = idx[cube_2d.coord_dims(x_name)[0]]
+        yi = idx[cube_2d.coord_dims(y_name)[0]]
+        x0, x1 = cube_2d.coord(x_name).bounds[xi]
+        y0, y1 = cube_2d.coord(y_name).bounds[yi]
+
+        if xmod:
+            x0, x1 = _rebase_values_to_modulus((x0, x1), xmod)
+        if ymod:
+            y0, y1 = _rebase_values_to_modulus((y0, y1), ymod)
+
+        # create a new polygon of the grid cell and check intersection
+        cell_box = sgeom.box(x0, y0, x1, y1)
+        intersect_bool = trans_geo.intersects(cell_box)
+        # mask all points without a intersection
+        if intersect_bool is False:
+            mask_template[idx] = True
+        # if weights method used, mask intersections below required weight
+        if intersect_bool is True and weights is True:
+            intersect_area = trans_geo.intersection(cell_box).area
+            if (intersect_area / cell_box.area) <= minimum_weight:
+                mask_template[idx] = True
+
+    return mask_template
+
+
+def _transform_coord_system(geometry, cube, geometry_system=None):
+    """Project the shape onto another coordinate system.
+
+    Arguments:
+        target: The target :class:`iris.coord_systems.CoordSystem`
+                or a :class:`iris.cube.Cube` object defining the coordinate
+                system to which the shape should be transformed
+
+    Returns:
+        A transformed shape (copy)
+    """
+    y_name, x_name = _cube_primary_xy_coord_names(cube)
+    DEFAULT_CS = _get_global_cs()
+    target_system = cube.coord_system()
+    if not target_system:
+        warnings.warn(
+            "Cube has no coord_system; using default GeogCS lat/lon",
+            IrisUserWarning,
+        )
+        target_system = DEFAULT_CS
+    if geometry_system is None:
+        geometry_system = DEFAULT_CS
+    target_proj = target_system.as_cartopy_projection()
+    source_crs = geometry_system.as_cartopy_projection()
+
+    trans_geometry = target_proj.project_geometry(geometry, source_crs)
+    if target_system == DEFAULT_CS and cube.coord(x_name).points[-1] > 180:
+        trans_geometry = shapely.transform(trans_geometry, _trans_func)
```

Comment:
Data which iris reads as DEFAULT_CS can have x_coords -180 to 180, or 0 to 360. Shapefile geometries (at least NE ones and most other true coordinates) are -180 to 180. Since the coordinate system is the same in iris for two different ranges of x coordinates, this line checks if the cube data for x coords goes beyond 180, and if so translates the geometry into 0-360 system. For rotated poles, they are a different coord_system than DEFAULT_CS, so the project_geometry line at 142 should catch them. It would be a good test case to add to the integrated tests though, as I have some data which is 360 starting x coords in a rotated pole which should be tested.